edifice/_component.py

```diff
@@ -522,8 +530,13 @@ def render(self):
         """
         raise NotImplementedError
 
+P = tp.ParamSpec("P")
+
+def not_ignored(arg: tuple[str, tp.Any]) -> bool:
+    return arg[0][0] != "_"
 
-def make_component(f):
+# TODO: Should we really allow the function to return `Any`?
+def component(f: Callable[tp.Concatenate[Component,P], tp.Any]) -> type[Component]:
```

Comment:
Why rename from `make_component` to `component`?

Reply:
Because I think this is going to be the main user-facing API and I want the ergonomics to be good.